README.md

```diff
@@ -17,76 +16,125 @@ _A table of contents to help people find things_
+_Note: at different viewport sizes, the message element hides the following elements:_
+- `<span class="o-message__content--detail">`
+- `<a class="o-message__action--secondary">`
 
-_Remember to start your codeblocks with three backticks and "html" so your markup is syntax highlighted correctly._
+A variation of the alert message is an **inner** alert message, which has almost exactly the same markup, with an optional addition of information, and does not have the option to close the message.
 
 ```html
-<div data-o-component="o-component-boilerplate" class='o-component-boilerplate'>
+<div class="o-message o-message--alert-inner o-message--success" data-o-component="o-message">
+	<div class="o-message__container">
+		<div class="o-message__content">
+			<p class="o-message__content--highlight">Hooray!
+				<span class="o-message__content--detail">The quick brown fox jumped over the lazy dogs!</span>
+			</p>
+			<p class="o-message__content--additional-info">Did you know that that sentence uses all of the letters in the alphabet at least once?</p>
+
+			<div class="o-message__actions">
+				<a href="#" class="o-message__action--primary">Button</a>
+				<a href="#" class="o-message__action--secondary">Text link</a>
+			</div>
+		</div>
+	</div>
 </div>
 ```
 
 ### JavaScript
-_Remember to start your codeblocks with three backticks and "js" so your js is syntax highlighted correctly._
-
-_Though it's not practical to repeat every aspect of Origami modules convention for every component, **A LOT** of people get tripped up by modules not auto initialising, so this line (remember to change the `o-component-boilerplate` to your component name) is useful if you have JavaScript:_
+No code will run automatically unless you are using the Build Service. You must either construct an `o-message` object or fire an o.DOMContentLoaded event, which `o-message` listens for. There are currently three variants of `o-message` that you can use: `alert`, `alert-bleed` and `alert-inner`. All of these variants require a status of either `success`, `error` or `neutral`.
 
-No code will run automatically unless you are using the Build Service.
-You must either construct an `o-component-boilerplate` object or fire the `o.DOMContentLoaded` event, which oComponent listens for.
 
-#### Constructing an o-component-boilerplate
+#### Construction
 
+If you have set up your message declaratively, and are using default o-message classes, use the following:
 ```js
-const oComponentBoilerplate = require('o-component-boilerplate');
-
-const oComponentBoilerplate = new oComponentBoilerplate();
+const oMessage = require('o-message');
+const messageElement = document.getElementById('my-message');
+const importantMessage = new oMessage(messageElement, {
+	type: 'alert',
+	status: 'success',
+	content: {
+		highlight: 'Hooray!'
+	}
+});
 ```
+The second argument that `oMessage` accepts is an [options object](#options), which can be used to change a message's markup, variant and style.
```

Comment:
A variant could include  markup and style changes. Maybe ", which can be used to change a message's style and functionality." But again I'm being pedantic -- apologies 🙈